runtime/container_linux.go

```diff
@@ -284,6 +284,52 @@ func (c *container) Stats() (*Stat, error) {
 	}, nil
 }
 
+// Status implements the runtime Container interface.
+func (c *container) Status() (State, error) {
+	args := c.runtimeArgs
+	args = append(args, "state", c.id)
+
+	out, err := exec.Command(c.runtime, args...).CombinedOutput()
+
+	// If only error happens, it is useful to return the detail.
+	if err != nil {
+		return "", fmt.Errorf(string(out))
+	}
+
+	// We may not know the exactly structure of the response if we
+	// have more runtimes other than OCI/runc.
+	var f interface{}
```

Comment:
Ah, that's a bad implementation indeed. Changed as your comments, it is more simple and clear now, thanks!